### DISCUSSION TITLE
Cannot use `xterm-ghostty`

### DISCUSSION THREAD

**Discussion by tangowithfoxtrot** [ghostty-org/ghostty#3534] [Q&A] [answered]
I'm not sure why, but every new session I open, I get the following error:

```
warning: Could not set up terminal.
warning: TERM environment variable set to xterm-ghostty
warning: Check that this terminal type is supported on this system.
warning: Using fallback terminal type xterm-256color
```

I found a similar issue: https://github.com/ghostty-org/ghostty/issues/586. But it was closed and appears to have been caused by something related to their developer setup. I have not built from source, so I don't believe that applies to my issue.

I can force the `term` config value to `term = xterm-256color`, which works around the error, but I would like to be able to use the image protocol for other things.

I'm using Fish shell `4.0.0-beta.1-gc473aa60a` on macOS Sequoia 15.2. I have also tried the latest stable version of Fish (`3.7.1`), so I don't believe it's my shell, but I could be wrong.

**Comment by mitchellh** [comment, marked as answer]
https://ghostty.org/docs/help/terminfo

**Reply by tangowithfoxtrot** [reply]
Apologies if I'm being dense, but I'm not sure how that answers it. I would _prefer_ to use `xterm-ghostty`, but if I let Ghostty use the default value for `term`, I get that error.

How do I prevent my session from having to fall back to `xterm-256color`?

**Reply by mitchellh** [reply]
This warning shows up when the terminfo can't be found, typically.

**Reply by rhodes-b** [reply]
> Apologies if I'm being dense, but I'm not sure how that answers it. I would _prefer_ to use `xterm-ghostty`, but if I let Ghostty use the default value for `term`, I get that error.
> 
> How do I prevent my session from having to fall back to `xterm-256color`?

Like the link to the doc explains you need to install terminfo on your machine. Im not familiar with macos but it should be bundled with the app

**Reply by rganesan** [reply]
I faced the same problem on Linux Mint. If you install ghostty from source in your home directory $HOME/.local, the terminfo is not automatically picked up by other applications. I worked around this by creating a symlink from $HOME/.terminfo to $HOME/.local/share/terminfo.

**Reply by tangowithfoxtrot** [reply]
Yeah. I later found that something was setting my `TERMINFO` variable on macOS to be that directory, where _Linux_ expects it to be. I was able to unset it so that it uses whatever macOS uses as the default for terminfo and can now use `xterm-ghostty`.

**Reply by jacobmischka** [reply]
For future searchers, my issue was that kitty created a `~/.terminfo` on my destination machine that I didn't realize, which didn't have ghostty's entries in it by default. Ghostty is installed on both machines, but the source machine is Linux and the destination is macOS.

I did `cp -R /Applications/Ghostty.app/Contents/Resources/terminfo/* ~/.terminfo` on the destination machine to fix for now. There's probably a better way but this was simplest for now, I couldn't get `TERMINFO` env var working via SSH.
